src/components/annotations/common_defaults.js

```diff
@@ -60,6 +60,10 @@ module.exports = function handleAnnotationCommonDefaults(annIn, annOut, fullLayo
 
         Lib.coerceFont(coerce, 'hoverlabel.font', {
             family: globalHoverLabel.font.family,
+            weight: globalHoverLabel.font.weight,
+            style: globalHoverLabel.font.style,
+            stretch: globalHoverLabel.font.stretch,
+            variant: globalHoverLabel.font.variant,
```

Comment:
I suspect it would be simpler to use `coerceFont` AND easier to read the logic if it accepted TWO default objects - the first letting you pass on an entire unchanged font object, the second letting you override just one or two parts. If we did that, this call would turn into something like
`Lib.coerceFont(coerce, ‘hoverlabel.font’, globalHoverLabel.font, {color: globalHoverLabel.font.color || hoverBorder})`